File: nextvisa-client/src/types/configuration.ts
export interface Configuration {
    id: number;
    base_url: string;
    hub_address: string;
    sleep_time: number;
    push_token: string;
    push_user: string;
    df_msg: string;
    created_at: string;
    updated_at?: string;
}

export interface ConfigurationUpdate {
    base_url: string;
    hub_address: string;
    sleep_time: number;
    push_token: string;
    push_user: string;
    df_msg: string;
}

File: nextvisa-api/models/configuration.py
from pydantic import BaseModel
from typing import Optional
from datetime import datetime

class ConfigurationBase(BaseModel):
    base_url: str
    hub_address: str
    sleep_time: float = 15.0
    push_token: str
    push_user: str
    df_msg: str

class ConfigurationCreate(ConfigurationBase):
    pass

class ConfigurationUpdate(ConfigurationBase):
    pass

class ConfigurationResponse(ConfigurationBase):
    id: int
    created_at: datetime
    updated_at: Optional[datetime] = None

    class Config:
        from_attributes = True